insights/specs/datasources/pcp.py

```diff
@@ -74,3 +91,38 @@ def pmlog_summary_args(broker):
         raise SkipComponent("Failed to get pmlogger metrics: {0}".format(str(e)))
 
     return "{0} {1}".format(pm_file, ' '.join(sorted(metrics)))
+
+
+@datasource(ros_collect, HostnameShort, HostContext)
+def pcp_raw_files(broker):
+    """
+    Return all existing paths of PCP RAW data files of yesterday.
+
+    .. note::
+        The "hostname" in the file paths will be removed by the "save_as"
+        in the `pcp_raw_data` spec.
+        All the collected PCP RAW data files will be stored in the
+        "var/log/pcp/pmlogger" directory in the archive.
+
+    Returns:
+        list(str): list of file path
+
+    Raises:
+        ContentException: Raises when number of PCP RAW files less than 3.
+    """
+    hostname = broker[HostnameShort].hostname
+    yesterday = (datetime.date.today() - datetime.timedelta(days=1)).strftime("%Y%m%d")
+
+    pm_root = os.path.join("/var/log/pcp/pmlogger/", hostname)
+
+    pm_files = glob.glob(os.path.join(pm_root, "{0}.*.xz".format(yesterday)))
+    pm_files.append(os.path.join(pm_root, "{0}.index".format(yesterday)))
+    pm_files = sorted(filter(lambda x: os.path.isfile(x), pm_files))
+
+    file_cnt = 0
+    for pf in pm_files:
```

Comment:
Nit: Maybe before looping through files we can check the length of pm_files, and raise exception if its below 3.
```suggestion
    if len(pm_files) < 3:
      raise ContentException("Incomplete PCP RAW data of {0}".format(yesterday))
    for pf in pm_files:
      file_cnt += 1 if pf.endswith(('.index', '.meta.xz', '.0.xz')) else 0
    if file_cnt < 3:
      raise ContentException("Incomplete PCP RAW data of {0}".format(yesterday))
```

Reply:
As there won't be too many files be found, I updated it into one line check,